src/sdl2_compat.c

```diff
@@ -456,19 +456,62 @@ static QuirkEntryType quirks[] = {
 };
 
 #ifdef __linux__
-static void OS_GetExeName(char *buf, const unsigned maxpath) {
+static void OS_GetExeName(char *buf, const unsigned maxpath, bool *use_base_path)
+{
     int ret;
     buf[0] = '\0';
     ret = readlink("/proc/self/exe", buf, maxpath);
     (void)ret;
+    if (strstr(buf, "/python")) {
+        /* Get the name of the script Python is running */
+        FILE *fp = fopen("/proc/self/cmdline", "r");
+        if (fp) {
+            char tmp[SDL2COMPAT_MAXPATH];
+            size_t len = fread(tmp, 1, sizeof(tmp) - 1, fp);
+            if (len > 0) {
+                char *spot = tmp, *end = tmp + len;
+                if (strstr(tmp, "python")) {
+                    /* Skip the name of the interpreter */
+                    while (spot < end) {
+                        if (*spot == '\0') {
+                            ++spot;
+                            break;
+                        }
+                        ++spot;
+                    }
+                }
+                if (spot < end) {
+                    /* Get the name of the script, including the directory containing it
+                     * e.g. /app/bin/src/tauon/__main__.py -> tauon/__main__.py
+                     */
+                    char *sep = strrchr(spot, '/');
+                    if (sep) {
+                        while (sep > spot) {
+                            --sep;
+                            if (*sep == '/') {
+                                ++sep;
+                                break;
+                            }
+                        }
+                        spot = sep;
+                    }
+                    strlcpy(buf, spot, maxpath);
```

Comment:
`strlcpy` isn't available everywhere, only recent glibc has it as I remember. Fails build for me, for e.g.

Reply:
Should I change it to `snprint(buf, maxpath, "%s", spot);` ?

Reply:
Can we use SDL3 functions at this point? If not, then yes please!

Reply:
> Can we use SDL3 functions at this point? If not, then yes please!

Not 100% sure, so I pushed the snprintf patch

That linux procedure uses several other libc calls which have SDL counterparts, strchr for e.g., so if you want to change to SDL calls, it won't be just for strlcpy.

Reply:
It looks like probably not ;)
```
/* DO NOT USE SDL3 FUNCTIONS IN HERE! */
```

Reply:
SDL2COMPAT_stpcpy() if we're still in dllinit, please.

Reply:
I'll fix that. Also rename it to SDL2COMPAT_strlcpy so it's more familiar with someone coming from SDL3 ;)

Reply:
er, or not. :)

Reply:
yeah, different interface.  :)

Reply:
> SDL2COMPAT_stpcpy() if we're still in dllinit, please.

Areyou sure? There is no buffer size check in that

Reply:
Also there are other libc calls in there, e.g str[r]chr, strstr, etc...

Reply:
> yeah, different interface. :)

Yup: https://linux.die.net/man/3/stpcpy

Reply:
Okay, let’s leave it as-is for now. We have bigger fish to fry 😊